### PR TITLE
[BugFix] agg output statistics use error row size

### PR DESCRIPTION
**Pull request by Seaven**
Fixes #issue

the join compute memCosts by row size, but row size is error in unknown statistics

like SQL:
`a2.a2`,  is `ARRAY<BIGINT>`, after statistics collect, the row size is 16.
`t0.v1`, is BIGINT, `array_agg(v1)` will produce `array<BIGINT>`, but the statistics is `UNKNOWN` and row size is 1.

```
MySQL td> explain costs select 1, a2.a2 from a2 join (select v2, array_agg(v1) from t0 group by v2) x0 on a2.v2 = x0.v2
+--------------------------------------------------------------------------+
| Explain String                                                           |
+--------------------------------------------------------------------------+
| PLAN FRAGMENT 0(F04)                                                     |
|   Output Exprs:14: expr | 5: a2                                          |
|   Input Partition: UNPARTITIONED                                         |
|   RESULT SINK                                                            |
|                                                                          |
|   7:EXCHANGE                                                             |
|      cardinality: 2                                                      |
|                                                                          |
| PLAN FRAGMENT 1(F03)                                                     |
|                                                                          |
|   Input Partition: HASH_PARTITIONED: 11: v2                              |
|   OutPut Partition: UNPARTITIONED                                        |
|   OutPut Exchange Id: 07                                                 |
|                                                                          |
|   6:Project                                                              |
|   |  output columns:                                                     |
|   |  5 <-> [5: a2, ARRAY<DECIMAL64(18,7)>, true]                         |
|   |  14 <-> 1                                                            |
|   |  cardinality: 2                                                      |
|   |  column statistics:                                                  |
|   |  * a2-->[-Infinity, Infinity, 0.0, 16.0, 1.0] ESTIMATE               |
|   |  * expr-->[1.0, 1.0, 0.0, 1.0, 1.0] ESTIMATE                         |
|   |                                                                      |
|   5:HASH JOIN                                                            |
|   |  join op: INNER JOIN (BUCKET_SHUFFLE(S))                             |
|   |  equal join conjunct: [2: v2, BIGINT, true] = [11: v2, BIGINT, true] |
|   |  build runtime filters:                                              |
|   |  - filter_id = 0, build_expr = (11: v2), remote = true               |
|   |  output columns: 5                                                   |
|   |  cardinality: 2                                                      |
|   |  column statistics:                                                  |
|   |  * v2-->[1.0, 2.0, 0.0, 8.0, 2.0] ESTIMATE                           |
|   |  * a2-->[-Infinity, Infinity, 0.0, 16.0, 1.0] ESTIMATE               |
|   |  * v2-->[1.0, 2.0, 0.0, 8.0, 2.0] ESTIMATE                           |
|   |  * expr-->[1.0, 1.0, 0.0, 1.0, 1.0] ESTIMATE                         |
|   |                                                                      |
|   |----4:AGGREGATE (update finalize)                                     |
|   |    |  group by: [11: v2, BIGINT, true]                               |
|   |    |  cardinality: 2                                                 |
|   |    |  column statistics:                                             |
|   |    |  * v2-->[1.0, 2.0, 0.0, 8.0, 2.0] ESTIMATE                      |
|   |    |                                                                 |
|   |    3:EXCHANGE                                                        |
|   |       distribution type: SHUFFLE                                     |
|   |       partition exprs: [11: v2, BIGINT, true]                        |
|   |       cardinality: 2                                                 |
|   |                                                                      |
|   1:EXCHANGE                                                             |
|      distribution type: SHUFFLE                                          |
|      partition exprs: [2: v2, BIGINT, true]                              |
|      cardinality: 3                                                      |
|      probe runtime filters:                                              |
|      - filter_id = 0, probe_expr = (2: v2)                               |
|                                                                          |
| PLAN FRAGMENT 2(F02)                                                     |
|                                                                          |
|   Input Partition: RANDOM                                                |
|   OutPut Partition: HASH_PARTITIONED: 11: v2                             |
|   OutPut Exchange Id: 03                                                 |
|                                                                          |
|   2:OlapScanNode                                                         |
|      table: t0, rollup: t0                                               |
|      preAggregation: on                                                  |
|      Predicates: 11: v2 IS NOT NULL                                      |
|      partitionsRatio=1/1, tabletsRatio=3/3                               |
|      tabletList=7304981,7304983,7304985                                  |
|      actualRows=2, avgRowSize=8.0                                        |
|      cardinality: 2                                                      |
|      column statistics:                                                  |
|      * v2-->[1.0, 2.0, 0.0, 8.0, 2.0] ESTIMATE                           |
|                                                                          |
| PLAN FRAGMENT 3(F00)                                                     |
|                                                                          |
|   Input Partition: RANDOM                                                |
|   OutPut Partition: HASH_PARTITIONED: 2: v2                              |
|   OutPut Exchange Id: 01                                                 |
|                                                                          |
|   0:OlapScanNode                                                         |
|      table: a2, rollup: a2                                               |
|      preAggregation: on                                                  |
|      Predicates: 2: v2 IS NOT NULL                                       |
|      partitionsRatio=1/1, tabletsRatio=3/3                               |
|      tabletList=7100669,7100671,7100673                                  |
|      actualRows=3, avgRowSize=24.0                                       |
|      Pruned type: 5 <-> [ARRAY<DECIMAL64(18,7)>]                         |
|      cardinality: 3                                                      |
|      probe runtime filters:                                              |
|      - filter_id = 0, probe_expr = (2: v2)                               |
|      column statistics:                                                  |
|      * v2-->[1.0, 3.0, 0.0, 8.0, 3.0] ESTIMATE                           |
|      * a2-->[-Infinity, Infinity, 0.0, 16.0, 1.0] ESTIMATE               |
+--------------------------------------------------------------------------+
91 rows in set
Time: 0.024s
MySQL td>
```

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java [hunk 1]
```diff
@@ -169,13 +169,19 @@ private double calculateKeySize() {
...
             } else {
                 Statistics.Builder allBuilder = Statistics.builder();
                 allBuilder.addColumnStatistics(rightTableStat.getColumnStatistics());
-                ColumnStatistic outputStatistic = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
-                keySize += outputStatistic.getAverageRowSize();
+                keyStatistics = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
+            }
+
+            if (keyStatistics.isUnknown()) {
+                keySize += keyStatistics.getAverageRowSize();
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java:182]: Why use averageRowSize when statistics is unknown?

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java [hunk 2]
```diff
@@ -169,13 +169,19 @@ private double calculateKeySize() {
...
+                keyStatistics = rightTableStat.getColumnStatistic((ColumnRefOperator) buildMapOp);
             } else {
                 Statistics.Builder allBuilder = Statistics.builder();
                 allBuilder.addColumnStatistics(rightTableStat.getColumnStatistics());
-                ColumnStatistic outputStatistic = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
-                keySize += outputStatistic.getAverageRowSize();
+                keyStatistics = ExpressionStatisticCalculator.calculate(buildMapOp, allBuilder.build());
+            }
+
+            if (keyStatistics.isUnknown()) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java:181]: Add comment for it

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java (hunk 1 above)
Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/HashJoinCostModel.java:182]: error code

**Comment by dengliu** [comment]
@chaoyli  @kangkaisen can this be back ported to 3.1?

**Comment by kangkaisen** [comment]
> @chaoyli @kangkaisen can this be back ported to 3.1?

@dengliu we will try to back port this PR to 3.1 branch

**Review by packy92 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**